src/pages/academy/gameSimulator/subcomponents/assetViewer/AssetViewer.tsx

```diff
@@ -0,0 +1,84 @@
+import { Icon, Tab, Tabs, Tooltip, Tree, TreeNodeInfo } from '@blueprintjs/core';
+import { cloneDeep } from 'lodash';
+import React from 'react';
+import { useRequest } from 'src/commons/utils/Hooks';
+import { fetchAssetPaths, s3AssetFolders } from 'src/features/gameSimulator/GameSimulatorService';
+import { deleteS3File } from 'src/features/gameSimulator/GameSimulatorService';
+
+import AssetViewerPreview from './AssetViewerPreview';
+import AssetViewerUpload from './AssetViewerUpload';
+import { convertAssetPathsToTree, treeMap } from './AssetViewerUtils';
+
+/**
+ * This component renders the Asset Viewer component in the Game Simulator.
+ *
+ * It provides a preview of all the S3 asset files in a document tree format.
+ * The selected asset will be available for preview.
+ */
+const AssetViewer = () => {
```

Comment:
Ditto on `React.FC`

Reply:
Addressed in [962b341](https://github.com/source-academy/frontend/pull/2836/commits/962b3413e413ff74052dd0e7f90ff5830ec709fb).